appcues/src/main/java/com/appcues/ui/extensions/ModifierExt.kt

```diff
@@ -130,31 +126,31 @@ internal fun Modifier.styleBorder(
     }
 )
 
-internal fun Modifier.styleSize(style: ComponentStyle, noSizeFillMax: Boolean = false) = this.then(
+internal fun Modifier.styleSize(style: ComponentStyle, contentMode: ComponentContentMode = ComponentContentMode.FIT) = this.then(
     when {
         // when style contains both width and height
         style.width != null && style.height != null -> when {
-            // we fill max if both properties are -1
-            style.width eq -1.0 && style.height eq -1.0 -> Modifier.fillMaxSize()
-            // or we set height and fill width in case only width is -1
+            // fill width in case only width is -1
             style.width eq -1.0 ->
                 Modifier
                     .height(style.height.dp)
                     .fillMaxWidth()
-            // or we set width and fill height in case only height is -1
-            style.height eq -1.0 ->
-                Modifier
-                    .width(style.width.dp)
-                    .fillMaxHeight()
             // else we set size with width and height
             else -> Modifier.size(style.width.dp, style.height.dp)
         }
         // if only width is not null, we fill max in case its -1 else we set the width
         style.width != null -> if (style.width eq -1.0) Modifier.fillMaxWidth() else Modifier.width(style.width.dp)
-        // if only height is not null, we fill max in case its -1 else we set the height
-        style.height != null -> if (style.height eq -1.0) Modifier.fillMaxHeight() else Modifier.height(style.height.dp)
-        // at the end we fill max size in case there is no width/height but the primitive (like image) is fill max by default
-        noSizeFillMax -> Modifier.fillMaxSize()
+        // if only height is not null, we set the height
+        style.height != null -> Modifier.height(style.height.dp).styleDefaultWidth(contentMode)
+        // at the end we fill max width if there is no width/height but the primitive (like image) is set to FILL
+        contentMode == ComponentContentMode.FILL -> Modifier.fillMaxWidth()
+        else -> Modifier
+    }
+)
+
+private fun Modifier.styleDefaultWidth(contentMode: ComponentContentMode) = this.then(
+    when (contentMode) {
+        ComponentContentMode.FILL -> Modifier.fillMaxWidth()
```

Comment:
this adds our full width for the case of only height specified, but a FILL image